routes/order.js

```diff
@@ -0,0 +1,115 @@
+var express = require('express');
+var router = express.Router();
+const { check, validationResult } = require('express-validator');
+const uuidv4 = require('uuid/v4');
+var body = require('express-validator'); //validation
+var moment = require('moment'); //datetime
+const multer = require('multer'); //middleware for handling multipart/form-data, which is primarily used for uploading files
+const upload = multer({ dest: './static/images/produce_images/' }); //path.join(__dirname, 'static/images/produce_images/)
+var ROLES = require('../utils/roles');
+var fs = require('fs');
+const axios = require('axios');
+const crypto = require('crypto');
+const hash = crypto.createHash('sha256');
+
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args));
+
+var initModels = require('../models/init-models');
+var sequelise = require('../config/db/db_sequelise');
+
+var models = initModels(sequelise);
+
+//GET order_dashboard
+router.get(
+  '/',
+  require('connect-ensure-login').ensureLoggedIn({ redirectTo: '/app/auth/login' }),
+  function (req, res, next) {
+    if (req.user.role === ROLES.Buyer || req.user.role === ROLES.Seller) {
+      models.Buyer_bid.findAll({
+        order: [['pk', 'DESC']],
+      })
+        .then(bid_rows => {
+          models.Seller_offer.findAll({
+            order: [['pk', 'DESC']],
+          }).then(offer_rows => {
+            produce_rows = [
+              'Baby Marrow',
+              'Baby Leeks',
+              'Basil',
+              'Beetroot',
+              'Bergamot',
+              'Blood Oranges',
+              'Cabbage',
+              'Carrots',
+              'Cauliflower',
+              'Cayenne Pepper',
+              'Cucumber',
+              'Eggs',
+              'Fennel',
+              'Granadilla',
+              'Green Beans',
+              'Herbs',
+              'Lebanese Cucumber',
+              'Leeks',
+              'Lemon',
+              'Lettuce',
+              'Limes',
+              'Mor',
+              'Onion',
+              'Pak Choi',
+              'Parsley',
+              'Radish',
+              'Sorrel',
+              'Swiss Chard',
+              'Spinach',
+              'Turnips',
+            ];
+            const produceArray = [];
+            if (bid_rows.length || offer_rows.length) {
+              for (var i = 0; i < produce_rows.length; i++) {
```

Comment:
lets make this more modular by creating a function that accepts the produce_rows, bid_rows and offer_rows and returns the finalProduceArray, then calling this function from inside your router.

i.e. something along the lines of - 
const finalProduceArray = []
`if (bid_rows.length || offer_rows.length) { 
finalProduceArray = returnRelevantProduce(produce_rows, bid_rows and offer_rows);
}

function returnRelevantProduce(produce_rows, bid_rows and offer_rows){
...
const relevantProduceArray = []
return relevantProduceArray = [...new Set(produceArray)];
}
`